adoptingerlang.org

```diff
@@ -3170,60 +3195,58 @@ In order to disable the scheduler busy waiting and to set the number of schedule
 +S ${SCHEDULERS}
 #+END_SRC
 
-In the next section we will see how using an environment variable =${SCHEDULERS}= makes it easy to adjust the number of schedulers when we want to change the CPU limits in the Kubernetes resource.
+In the next section we will see how using an environment variable =${SCHEDULERS}= set through Kubernetes' [[https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#the-downward-api][Downward API]] makes it easy to adjust the number of schedulers when we want to change the CPU limits in the Kubernetes resource.
 
 **** Container Environments and ConfigMaps
 
-As we've seen in the Releases chapter, runtime configuration is done through environment variable substitution in =vm.args.src= and =sys.config.src=. We therefore have to insert those variable to the environment of the container. Each container in a Kubernetes Pod can have a field =env= declaring a set of environment variables:
-
-[TODO: figure out using dowardAPI for setting schedulers]
+As we've seen in the Releases chapter, runtime configuration is done through environment variable substitution in =vm.args.src= and =sys.config.src=. We therefore have to insert those variable to the environment of the container. Each container in a Kubernetes Pod can have a field =env= declaring a set of environment variables. The simplest case is explicitly providing a =name= and a =value=:
 
 #+BEGIN_SRC yaml
 env:
 - name: LOGGER_LEVEL
   value: error
-- name: PORT
-  value: 8053
-- name: SCHEDULERS
-  value: 2
 #+END_SRC
 
+This configuration would result in the environment variable =LOGGER_LEVEL= with value =error=.
+
+There are other environment variables that must be set based on the state of the container that is running. Two examples of this are setting the =NODE_IP= variable based on the IP of the Pod and setting the =SCHEDULERS= based on the CPU limits using Kubernetes' [[https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#the-downward-api][Downward API]]:
+
 #+BEGIN_SRC yaml
 env:
 - name: NODE_IP
   valueFrom:
     fieldRef:
       fieldPath: status.podIP
+- name: SCHEDULERS
+  valueFrom:
+    resourceFieldRef:
+      containerName: service-discovery
+      resource: limits.cpu
+      divisor: 1
 #+END_SRC
 
+The =fieldRef= =status.podIP= will return the current Pod's IP when it is created and the =resourceFieldRef= of =limits.cpu= with a =divisor= of =1= will return the CPU limit in a unit of CPU cores.
```

Comment:
is this intended to say `The =fieldRef='s =status.podIP=` or something? It's a bit unclear. Or maybe `The =status.podIP= declaration under =fieldRef= will return ...`

Reply:
The latter. I've pushed an update.